xtask/src/flash_image.rs

```diff
@@ -151,8 +118,8 @@ impl FlashImage {
         // Parse and verify checksums
         let header_checksum = u32::from_le_bytes(image[8..12].try_into().unwrap());
         let payload_checksum = u32::from_le_bytes(image[12..16].try_into().unwrap());
-        let calculated_header_checksum = FlashImageChecksum::calculate_checksum(&image[0..8]);
-        let calculated_payload_checksum = FlashImageChecksum::calculate_checksum(&image[16..]);
+        let calculated_header_checksum = calculate_checksum(&image[0..8]);
+        let calculated_payload_checksum = calculate_checksum(&image[16..]);
```

Comment:
If these were parsed with `zerocopy` you could just use their `calculate_checksum()` methods.

Reply:
sounds good. i can definitely parse these using zerocopy, although the payload itself can not be parsed since it has variable length.